src/actor/npc/ac_npc_shop_master.c

```diff
@@ -0,0 +1,2677 @@
+    if (pswd->hit_rate_index == 1) {
+        if (mMpswd_check_name(pswd) == TRUE) {
+            res = aNSC_PSW_GOOD_USER;
+        } else {
+            res = aNSC_PSW_BAD;
+        }
+    }
+    return res;
+}
+
+static int aNSC_pc_check_password(NPC_SHOP_MASTER_ACTOR *shop_master) {
+    static int (*pc_check_password_proc[6])(mMpswd_password_c *) = {
+        aNSC_pc_check_password_famicom,
+        aNSC_pc_check_password_npc,
+        aNSC_pc_check_password_card_e,
+        aNSC_pc_check_password_magazine,
+        aNSC_pc_check_password_card_e_mini,
+        aNSC_pc_check_password_user
+    };
+
+    int res = aNSC_PSW_RES_0; 
+    u8 buff[24];
+
+    if (shop_master->password_str[mMpswd_PASSWORD_STR_LEN - 1] == ' ') {
+        res = aNSC_PSW_RES_9;
+    } else {
+        if (mMpswd_decode_code(buff, shop_master->password_str) == TRUE) {
+            mMpswd_password_c *password = &shop_master->password;
+
+            mMpswd_password(buff, password);
+            if (mMpswd_password_zuru_check(password) == FALSE && mMpswd_check_present(password) == TRUE) {
+                aNSC_set_pw_password_str(shop_master);
+                aNSC_set_pw_info_str(shop_master);
+                res = pc_check_password_proc[password->type](password);
+            }
+        }
+    }
+
+    return res;
+}
+
+static int aNSC_check_possession_item_make_password(NPC_SHOP_MASTER_ACTOR *shop_master) {
+    mActor_name_t *item;
+    int flags;
+    int i;
+    int res;
+
+    flags = 0;
+    res = 0;
+
+    item = Now_Private->inventory.pockets;
+    for (i = 0; i < 15; i++) {
+        if ((*item != EMPTY_NO) && (mPr_GET_ITEM_COND(Now_Private->inventory.item_conditions, i) == 0) && mMpswd_check_present_user(*item) == TRUE) {
+            res = 1;
+            flags |= (1 << i);
+        }
+        item += 1;
+    }
+    shop_master->password_flags = flags; 
+
+    return res;
+}
+
+static void aNSC_set_buy_sum_str(mActor_name_t item, u32 p2) {
+    static int tanni_string_num[10] = {0x566, 0x575, 0x584, 0x593, 0x5a2, 0x5b1, 0x5c0, 0x5cf, 0x5de, 0x5ed};
+    mActor_name_t type;
+    u8 buff[16];
+    u32 str_no;
+
+    aNSC_set_item_name_str(item, mMsg_ITEM_STR2);
+    aNSC_set_value_str(p2, 7);
+
+    switch (ITEM_NAME_GET_TYPE(item)) {
+    case NAME_TYPE_ITEM1:
+        str_no = p2 + tanni_string_num[aNSC_item_tanni_type[ITEM_NAME_GET_CAT((mActor_name_t)item)][ITEM_NAME_GET_INDEX(item)]];
+        str_no -= 1;
+        break;
+    case NAME_TYPE_FTR0:
+    case NAME_TYPE_FTR1:
+        str_no = p2 + 0x565;
+        break;
+    default:
+        str_no = p2 + 0x565;
+        break;
+    }
+
+    mString_Load_StringFromRom(buff, 0x10, str_no);
+    mMsg_Set_free_str(mMsg_Get_base_window_p(), 8, buff, 0x10);
+}
+
+static int aNSC_decide_next_move_act(NPC_SHOP_MASTER_ACTOR *shop_master, GAME_PLAY *play) {
+    int res;
+    PLAYER_ACTOR *player;
+    if ((shop_master->npc_class.collision.pipe.collision_obj.collision_flags0 & 0x2) != 0) {
+        return 0x3d;
```

Comment:
These should also be turned into action enums, same one that gets passed into aNSC_setupAction likely.